### PR TITLE
Add shader package

### PR DESCRIPTION
**Pull request by riknoll**
This adds a new package called shader that has some useful functions for shading:

1. mapImage - similar to mapRect except that it remaps colors according to an image. It takes in a buffer of color mappings and each color in the map image will point to a different section of the buffer, allowing you to shade the destination image by different amounts.
2. mergeImage - takes two images and finds the minimum pixel value at each coordinate. Useful for merging multiple light sources into one "lightSpace" that you can shade all at once

### CODE REVIEW

**Review by jwunderl — APPROVED** [review]

File: libs/shader/sim/shader.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+/// <reference path="../../screen/sim/image.ts" />
+
+namespace pxsim.ShaderMethods {
+    export function _mergeImage(dst: RefImage, src: RefImage, xy: number) {
+        mergeImage(dst, src, pxsim.ImageMethods.XX(xy), pxsim.ImageMethods.YY(xy));
+    }
+
+    function mergeImage(dst: RefImage, src: RefImage, x0: number, y0: number) {
```
Comment (eanders-ms) [inline libs/shader/sim/shader.ts:8]: Just wondering, why reimplement this? Could you call `pxsim.ImageMethods.mergeImage/mapImage` here instead?

**Review by eanders-ms — APPROVED** [review]

File: libs/shader/sim/shader.ts (hunk 1 above)
Comment (riknoll) [reply libs/shader/sim/shader.ts:8]: this is the simulator implementation!

Comment (riknoll) [reply libs/shader/sim/shader.ts:8]: these apis are the ones i'm adding, so they don't already exist in imagemethods

Comment (eanders-ms) [reply libs/shader/sim/shader.ts:8]: Would this be equivalent?
```ts
    function mergeImage(dst: RefImage, src: RefImage, x0: number, y0: number) {
        pxsim.ImageMethods.mergeImage(dst, src, x0, y0);
    }
```

**Pull request merged**